### ISSUE TITLE
Potential UI Bug when adding/updating JSON fields in Admin Panel

### ISSUE THREAD

**Issue by wrobson-lllow** [payloadcms/payload#4403]
### Describe the Bug

We recently updated our application from payload ^1.8.2 to payload to ^2.3.0. Since updating we have been experiencing a peculiar issue with the JSON fields, it's hard for me to pinpoint exactly when and why it happens, but it seems to happen more often than it does not.  When entering info into the JSON field (which has a default value of {}) the cursor will jump to the end of the field. It seems to happen every time the field automatically formats, however at times it also happens on any keystroke. Once it happens once it seems to continue to happen on every keystroke.

This behavior has been reproduced on several machines both locally and on our deployed website.

I'm including screenshots, however when I take a screenshot the JSON field loses focus and I am unable to show the cursor in the screenshot. The red arrows represent where the cursor would be.

Is the JSON field using the Slate Editor or the Monaco editor? Are there any config options that I am failing to include?

### To Reproduce

Starting with an empty field (default value: {})
![image](https://github.com/payloadcms/payload/assets/132296416/3916d919-5bd1-4d28-a0b2-ece896131680)
The first key can be inputted as expected:
![image](https://github.com/payloadcms/payload/assets/132296416/b2752f43-922d-4579-aa68-7f0e7232ff7c)
When a value is entered for the key the cursor jumps to the end
![image](https://github.com/payloadcms/payload/assets/132296416/4dc36db1-0b5d-4955-9366-45b81b41f9c6)
Click back to where I was last:
![image](https://github.com/payloadcms/payload/assets/132296416/27e0a43d-a8fa-4684-8603-07990f82c4b2)
Add quotes for a string and the cursor jumps to the end again:
![image](https://github.com/payloadcms/payload/assets/132296416/21db306d-8e18-4246-ba9f-0e75748e05f7)
Click back to the quotes:
![image](https://github.com/payloadcms/payload/assets/132296416/c776d6cf-1a16-4055-8599-cef2a82d85b8)
Start typing and first character is placed correctly then cursor jumps to end where the rest of the text is:
![image](https://github.com/payloadcms/payload/assets/132296416/99ad716b-3fce-42be-8e0e-55f917557e7e)
Backspace also causes the same behavior.

Here's a screenshot of my payload.config where I have the editor defined:
![image](https://github.com/payloadcms/payload/assets/132296416/2b32f8ec-61c0-4f6b-b7bf-532582af162a)

### Payload Version

^2.3.0

### Adapters and Plugins

db-mongodb, bundler-webpack. editor-slateEditor

**Comment by wrobson-lllow** [comment]
Just to add, I'm getting a compile error: "Collection "server-services" > Field "value" does not match any of the allowed types" when I add "editorOptions" to the admin of the JSON field:
![image](https://github.com/payloadcms/payload/assets/132296416/78cc5a79-c0f7-46f2-af9a-aaba4ccfda64)

**Comment by jschuur** [comment]
> Just to add, I'm getting a compile error: "Collection "server-services" > Field "value" does not match any of the allowed types" when I add "editorOptions" to the admin of the JSON field

Were you able to resolve this? Similar problem was also reported [for the JSON field type](https://github.com/payloadcms/payload/issues/2729).

I'm having an issue where using useField's `setValue` to update a JSON field didn't show the new content, and was hoping this would force it to to refresh:

```js
    {
      name: 'content',
      type: 'json',
      defaultValue: defaultSurveyContent,
      admin: {
        editorOptions: {
          formatOnPaste: true,
          formatOnType: true,
        }
      }
    }
```

**Comment by wrobson-lllow** [comment]
A bug fix for my issue (not sure if it would solve yours) was released in version 2.5.0

**Comment by jschuur** [comment]
> A bug fix for my issue (not sure if it would solve yours) was released in version 2.5.0

Turns out I was still on 2.0.0 (fresh create-payload-app setup), but upgrading to 2.5.0 didn't make a difference.
